error_handling/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/blocky/as-demo/as"
+)
+
+type Result struct {
+	Success bool
+	Value   any
+}
+
+func (r Result) jsonMarshalWithError(err error) []byte {
+	resultStr := fmt.Sprintf(`{ "Success": false, "Value": "%v" }`, err)
```

Comment:
While `err` could be `nil`, the `Sprintf` does not panic and instead serialized `nil` error as "<nil>"